core/embed/rust/src/ui/model_mercury/layout.rs

```diff
@@ -1026,14 +1018,104 @@ extern "C" fn new_select_word(n_args: usize, args: *const Obj, kwargs: *mut Map)
     unsafe { util::try_with_args_and_kwargs(n_args, args, kwargs, block) }
 }
 
+#[derive(Copy, Clone, PartialEq, Eq, ToPrimitive)]
+pub enum SetBrightnessAction {
+    Slider,
+    Menu,
+    Confirm,
+}
+
+impl FlowState for SetBrightnessAction {
+    fn handle_swipe(&self, direction: SwipeDirection) -> Decision<Self> {
+        match (self, direction) {
+            (SetBrightnessAction::Menu, SwipeDirection::Right) => {
+                Decision::Goto(SetBrightnessAction::Slider, direction)
+            }
+            (SetBrightnessAction::Slider, SwipeDirection::Up) => {
+                Decision::Goto(SetBrightnessAction::Confirm, direction)
+            }
+            (SetBrightnessAction::Confirm, SwipeDirection::Down) => {
+                Decision::Goto(SetBrightnessAction::Slider, direction)
+            }
+            (SetBrightnessAction::Confirm, SwipeDirection::Left) => {
+                Decision::Goto(SetBrightnessAction::Menu, direction)
+            }
+            _ => Decision::Nothing,
+        }
+    }
+
+    fn handle_event(&self, msg: FlowMsg) -> Decision<Self> {
+        match (self, msg) {
+            (SetBrightnessAction::Slider, FlowMsg::Info) => {
+                Decision::Goto(SetBrightnessAction::Menu, SwipeDirection::Left)
+            }
+            (SetBrightnessAction::Menu, FlowMsg::Cancelled) => {
+                Decision::Goto(SetBrightnessAction::Slider, SwipeDirection::Right)
+            }
+            (SetBrightnessAction::Menu, FlowMsg::Choice(0)) => Decision::Return(FlowMsg::Cancelled),
+            (SetBrightnessAction::Confirm, FlowMsg::Confirmed) => Decision::Return(FlowMsg::Confirmed),
+            (SetBrightnessAction::Confirm, FlowMsg::Info) => {
+                Decision::Goto(SetBrightnessAction::Menu, SwipeDirection::Left)
+            }
+            _ => Decision::Nothing,
+        }
+    }
+}
+
+static mut BRIGHTNESS: u16 = 0;
```

Comment:
I also suggest indicating that `BRIGHTNESS` is in values of `[0-100] %`.

Either by a short commentary above it or changing the name to `BRIGTHNESS_PCT`

Reply:
> I thing we can avoid the unsafe by using `core::sync::atomic::AtomicU16` instead.

Indeed! Thank you!

https://github.com/trezor/trezor-firmware/pull/3991/commits/9cfa654b31ce9528782e4506c003b5b799d3b7e0#diff-369a97745e9678828b91bb11d8de987cfac308ea29ed1702009e1f58d16fd45aR72

Reply:
> I also suggest indicating that `BRIGHTNESS` is in values of `[0-100] %`.
> 
> Either by a short commentary above it or changing the name to `BRIGTHNESS_PCT`

Actually this is `[0..255]`, not a percentage. It gets converted to percentage when displayed. But thanks to your comment I realized we should be just using u8 rather than u16 everywhere related to brightness. I'll make a separate PR on top of this.